theano/sandbox/cuda/dnn.py

```diff
@@ -2246,9 +2247,13 @@ def local_dnn_conv_inplace(node):
         inputs = list(node.inputs)
         dest = inputs[2]
         if (dest.owner and
-                isinstance(dest.owner.op, GpuAllocEmpty) and
+                type(dest.owner.op) is GpuAllocEmpty and
```

Comment:
In the new back-end, we use type equality. We don't want subclass to get optimized by default. We prefer them to tell explicitly they want to be optimized. This is safer (we prefer to risk less optimized graph then bugged graph). Doing that here isn't a problem I think.